ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/MiniProtocol/ChainSync/Client.hs

```diff
@@ -1204,7 +1204,12 @@ checkTime cfgEnv intEnv =
         Intersects kis2 lst        <- checkArrivalTime kis arrival
         Intersects kis3 ledgerView <- case projectLedgerView slotNo lst of
             Just ledgerView -> pure $ Intersects kis2 ledgerView
-            Nothing         -> readLedgerState kis2 (projectLedgerView slotNo)
+            Nothing         -> do
+              let forecastAtSlot = forecastAt (ledgerViewForecastAt (configLedger cfg) lst)
+              EarlyExit.lift $ traceWith (tracer cfgEnv) $ TraceWaitingBeyondForecastHorizon forecastAtSlot slotNo
+              res <- readLedgerState kis2 (projectLedgerView slotNo)
```

Comment:
Hmm, we are now calling `ledgerViewForecastAt` again just for tracing :thinking: With a bit of refactoring, it should be possible to reuse the result from `projectLedgerView`. We could also keep it like this for now, and create an action item to change this later (or even just decide to remove these trace events if they are only useful for local debugging).

Also: style nitpick: line length

Reply:
To be fair, I am not sure how necessary it is to trace where the forecast is taken from, now that we know that forecasting seems to be working fine, it was mostly to debug forecasting itself. I will try to think of a solution to either get rid of this completely or at least avoid computing things twice.

Reply:
I removed it altogether and fixed the style in f7ed01f4c386413e546dfce66123b5738269c21b.